### PR TITLE
add properties for nfs extended groups

### PR DESCRIPTION
**Pull request by mannywarrior**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by brpanask** [comment]
Do you have example file updates to make for this?
Get/List/ListByRG/Update/CreateorUpdate

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 1]
```diff
@@ -1368,6 +1372,79 @@
         }
       }
     },
+    "CacheDirectorySettings": {
+      "description": "Cache Directory Services settings.",
+      "properties": {
+        "usernameDownload" : {
+          "$ref": "#/definitions/CacheUsernameDownloadSettings",
```
Comment (brpanask) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1387]: Are more Properties expected to be coming in the future outside of the username download settings?
Just asking in case the fields from username download settings could go directly into the Cache directory settings.

We have two conflicting examples above this, so it can go either way.
I see cache encryption settings which just goes on to reference another definition and has no other fields.
Then below that is cache security settings which directly has fields in it.

Comment (mannywarrior) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1387]: yes. The SMB related settings will be another property of Cache Directory Services settings. I thought it'd be better to have additional definitions for those settings.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "swagger": "2.0",
   "info": {
-    "description": "A Storage Cache provides scalable caching service for NAS clients, serving data from either NFSv3 or Blob at-rest storage (referred to as \"Storage Targets\"). These operations allow you to manage Caches.",
+    "description": "A Storage Cache provides scalable caching service for NAS clients, serving data from either NFSv4 or Blob at-rest storage (referred to as \"Storage Targets\"). These operations allow you to manage Caches.",
```
Comment (mannywarrior) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Not sure why this change shows in here. Initially I thought it was due to a merge, but I don't see any changes other than my commit that changed this

**Comment by nydickmsft** [comment]
The token “NFS3” was changed to “NFS4”.   That looks like a mistake.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 3]
```diff
@@ -1368,6 +1372,161 @@
...
+    },
+    "CacheActiveDirectorySettings": {
+      "description": "Active Directory settings used to join a Cache to a Domain.",
+      "properties": {
+        "primaryDnsIpAddress": {
+          "description": "primary DNS IP address of the AD Domain controller.",
+          "type": "string"
+        },
+        "secondaryDnsIpAddress": {
+          "description": "secondary DNS IP address of the AD Domian controller.",
```
Comment (brpanask) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Domian

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 4]
```diff
@@ -1368,6 +1372,161 @@
         }
       }
     },
+    "CacheDirectorySettings": {
+      "description": "Cache Directory Services settings.",
+      "properties": {
```
Comment (brpanask) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1377]: type: object

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 5]
```diff
@@ -1368,6 +1372,161 @@
...
+            "No"
+          ],
+          "x-ms-enum": {
+            "name": "DomainJoinedType",
+            "modelAsString": true
+          }
+        },
+        "credentials": {
+          "description": "Active Directory admin or user credentials used to join the HPC Cache to a Domain.",
+          "properties": {
```
Comment (brpanask) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1431]: type: object
